ui/app/components/app/whats-new-popup/whats-new-popup.js

```diff
@@ -0,0 +1,180 @@
+    </div>
+  );
+};
+
+const renderSubsequentNotification = (notification, idRefMap) => {
+  const { id, date, title, description, actionText } = notification;
+
+  const actionFunction = getActionFunctionById(id);
+  return (
+    <div
+      className={classnames('whats-new-popup__notification')}
+      key={`whats-new-popop-notificatiion-${id}`}
+      ref={idRefMap[id]}
+    >
+      <div className="whats-new-popup__notification-title">{title}</div>
+      <div className="whats-new-popup__description-and-date">
+        <div className="whats-new-popup__notification-description">
+          {description}
+        </div>
+        <div className="whats-new-popup__notification-date">{date}</div>
+      </div>
+      {actionText && (
+        <div className="whats-new-popup__link" onClick={actionFunction}>
+          {actionText}
+        </div>
+      )}
+    </div>
+  );
+};
+
+export default function WhatsNewPopup({ onClose }) {
+  const t = useContext(I18nContext);
+
+  const notifications = useSelector(getSortedNotificationsToShow);
+
+  const [seenNotifications, setSeenNotifications] = useState({});
+
+  const popoverRef = useRef();
+
+  const memoizedNotifications = useEqualityCheck(notifications);
+  const idRefMap = useMemo(
+    () =>
+      memoizedNotifications.reduce(
+        (_idRefMap, notification) => ({
+          ..._idRefMap,
+          [notification.id]: React.createRef(),
+        }),
+        {},
+      ),
+    [memoizedNotifications],
+  );
+
+  useEffect(() => {
+    const observer = new window.IntersectionObserver(
+      (entries, _observer) => {
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            const [id, ref] = Object.entries(idRefMap).find(([_, _ref]) =>
+              _ref.current.isSameNode(entry.target),
+            );
+
+            setSeenNotifications((_seenNotifications) => ({
+              ..._seenNotifications,
+              [id]: true,
+            }));
+
+            _observer.unobserve(ref.current);
+          }
+        });
+      },
+      {
+        root: popoverRef.current,
+        threshold: 1.0,
+      },
+    );
+
+    Object.values(idRefMap).forEach((ref) => {
+      observer.observe(ref.current);
+    });
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [idRefMap, setSeenNotifications]);
+
+  return (
+    <Popover
+      className="whats-new-popup__popover"
+      title={t('whatsNew')}
+      onClose={() => {
+        updateViewedNotifications(seenNotifications);
+        onClose();
+      }}
+      popoverRef={popoverRef}
+      mediumHeight
+    >
+      <div className="whats-new-popup__notifications">
+        {notifications.map(({ id }, index) => {
+          const notification = getTranslatedUINoficiations(t)[id];
+          return index === 0
+            ? renderFirstNotification(notification, idRefMap)
+            : renderSubsequentNotification({
+                notification,
+                idRefMap,
+              });
```

Comment:
Looks like you forgot to get rid of the curly braces here:

```suggestion
            : renderSubsequentNotification(
                notification,
                idRefMap,
              );
```

Reply:
thanks... this makes me want to do a final QA pass, which I will do now